### ISSUE TITLE
Yolo 26 ONNX and TFLITE segmentation mask size (resolution)

### ISSUE THREAD

**Issue by master-divanov** [ultralytics/ultralytics#23820]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

We are using Yolo 26 with image size imgsz=256, masks_ratio=1, We expect to see result in ONNX and TFLITE for segmentation mask to be 256x256, we are seeing 64x64 that seems to correspond to default settings masks_ratio=4

### Environment

(base) crane@crane-NUCXI7:~$ yolo checks
Ultralytics 8.4.18 🚀 Python-3.11.14 torch-2.10.0+cu128 CUDA:0 (NVIDIA GeForce RTX 3070 Laptop GPU, 7842MiB)
Setup complete ✅ (16 CPUs, 62.4 GB RAM, 2657.5/3753.4 GB disk)

 

OS                     Linux-6.17.0-14-generic-x86_64-with-glibc2.39
Environment            Linux
Python                 3.11.14
Install                pip
Path                   /home/crane/miniconda3/lib/python3.11/site-packages/ultralytics
RAM                    62.45 GB
Disk                   2657.5/3753.4 GB
CPU                    11th Gen Intel Core i7-11800H @ 2.30GHz
CPU count              16
GPU                    NVIDIA GeForce RTX 3070 Laptop GPU, 7842MiB
GPU count              1
CUDA                   12.8

 

numpy                  ✅ 2.1.3>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.13.0.92>=4.6.0
pillow                 ✅ 12.1.1>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.17.0>=1.4.1
torch                  ✅ 2.10.0>=1.8.0
torch                  ✅ 2.10.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.25.0>=0.9.0
psutil                 ✅ 7.2.2>=5.8.0
polars                 ✅ 1.38.1>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18

 

### Minimal Reproducible Example

import os

from ultralytics import YOLO

from ultralytics.utils import tuner

from ultralytics import RTDETR

 

model = YOLO('yolo26n-seg.yaml').load('yolo26n-seg.pt')  

results = model. Train(resume=True, data="./data.yaml", mask_ratio=1, device=0, imgsz=256, epochs=1000, batch=32)

 

### Additional

Ultralytics uses mask_ratio when rasterizing polygon labels into training masks. With imgsz=256 and mask_ratio=1, the GT mask canvas is 256×256. Because your run also has overlap_mask=True, it stores one overlapped index mask per image and reconstructs individual instance masks later inside the loss. 

The model’s raw instance-mask prototypes are set by the head architecture. Segment26 calls Proto26, and Proto26 calls Proto, which has only one ×2 transpose-convolution upsample. In your graph, the first segmentation feature map is the stride-8 map, so for a 256 input it is 32×32; one ×2 upsample makes the raw proto tensor 64×64. That is why mask_ratio=1 does not make the raw proto 256×256

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @master-divanov, thanks for the detailed report and for using Ultralytics 🚀 This is an automated response to help move things forward quickly; an Ultralytics engineer will also assist you soon.

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) for relevant [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, as many common workflows are covered there.

Since this is a 🐛 Bug Report, please share a fully runnable [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) focused specifically on:
-Training command/script
-Export command(s) for ONNX and TFLite
-Inference snippet showing the observed output mask shape
-Expected vs actual mask resolution from both exported formats
This helps us reliably reproduce and triage the behavior 🔍

If this is also related to custom training ❓, please include dataset image examples, complete training/export logs, and confirm alignment with [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify this is not already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
This is the expected behavior. `mask_ratio` does not control the output resolution of the mask. The output resolution is currently hardcoded to 1/4th the original image size.
